### PR TITLE
payslip API - 26 October 2022

### PR DESCRIPTION
**Pull request by nameless0724**

### CODE REVIEW

File: back_end/express.js [hunk 1]
```diff
@@ -136,14 +136,53 @@ app.post('/newprofile', async (req, res) => {
...
     } catch (error) {
         console.error(error.message);
         res.status(500).send({
             msg: "Unauthenticated"
         })
     }
 })
+
+//for payslip form
+app.post('/entrypayslip', async (req, res) => {
```
Comment (aeoxia) [inline back_end/express.js]: change to /payslip

**Review by aeoxia — CHANGES_REQUESTED** [review]
Apply comments hehe

File: back_end/express.js [hunk 2]
```diff
@@ -136,14 +136,53 @@ app.post('/newprofile', async (req, res) => {
...
+            user,
+            salary,
+            sss,
+            philhealth,
+            pagibig,
+            tax,
+            netpay
+        } = req.body
+
+        pool.query(`
```
Comment (aeoxia) [inline back_end/express.js]: const payslip = pool.query( ........ query here )

File: back_end/express.js [hunk 3]
```diff
@@ -136,14 +136,53 @@ app.post('/newprofile', async (req, res) => {
...
+        INSERT INTO public.payslip (id, "user", salary, sss, philhealth, pagibig, tax, netpay) VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING *
+        `, [uuidv4(), user, salary, sss, philhealth, pagibig, tax, netpay])
+
+        const message = "Entry complete!"
+
+        res.json({
+            message
```
Comment (aeoxia) [inline back_end/express.js]: payslip.rows[0]

Comment (nameless0724) [reply back_end/express.js]: Sir, this gives "cannot read properties of undefined (reading '0')" in the program.

**Review by aeoxia — APPROVED** [review]

**Pull request merged**